scenarios/helpers/config.erl

```diff
@@ -0,0 +1,100 @@
+%%%-------------------------------------------------------------------
+%%% @author denys
+%%% @copyright (C) 2019, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 30. Mar 2019 01:15
+%%%-------------------------------------------------------------------
+-module(config).
+-author("denys").
+
+%% API
+-export([get_scenario_settings/1, store_scenario_settings/1, get_parameter/1, dump_settings/0]).
+
+-export_type([scenario_configuration/0, settings/0]).
+
+-type name() :: atom().
+-type env_var() :: atom().
+-type value() :: any().
+
+-type one_of() :: [any()].
+-type func() :: fun((any()) -> boolean()).
+-type validation() :: func() | one_of() | bitstring | nonnegative_integer | positive_integer | none.
+-type reason() :: atom().
+
+-type parameter_configuration() :: {name(), env_var(), DefValue :: value(), validation()}.
+-type scenario_configuration() :: [parameter_configuration()].
+
+-type settings() :: [{name(), value()}].
+-type invalid_settings() :: {invalid_settings, [{name(), value(), reason()}]}.
+
+-spec get_scenario_settings(scenario_configuration()) -> {ok, settings()} | {error, invalid_settings()}.
+get_scenario_settings(Config) ->
+    ParametersVerification = [get_value_and_verify(C) || C <- Config],
+    case [{N, V, R} || {N, V, {false, R}} <- ParametersVerification] of
+        [] -> {ok, [{Name, Value} || {Name, Value, _} <- ParametersVerification]};
+        InvalidSettings -> {error, {invalid_settings, InvalidSettings}}
+    end.
+
+-spec store_scenario_settings(settings()) -> any().
+store_scenario_settings(Settings) ->
+    erlang:put(scenario_settings, Settings).
```

Comment:
it's not bad idea and i was also thinking about it, since the config is common for all the users. but the main question is which process should be the owner of such ETS table? `init` process dies, client/user processes can die and that can be absolutely normal scenario flow (e.g. GDPR removal testing)

Reply:
How about coordinator process or coordinator spawn_links process that owns the table and waits forever for a message (basically doing nothing)?

Reply:
the key point is that i don't know how to make it right as the separate helper module. 

you never know if coordinator process is available for the scenario, if it's running till the end or ends right after coordination is done. i don't want to do any assumptions about the scenario in the helper module. 

the only option i see is to create the new separate process, but it's a bit too much.